esp/logging/logginglib/loggingagentbase.cpp

```diff
@@ -422,6 +384,42 @@ String* CLogAgentBase::CVariant::normalize(const char* token)
     return tmp.getClear();
 }
 
+CEspLogAgentVariantIterator::CEspLogAgentVariantIterator()
+{
+}
+
+CEspLogAgentVariantIterator::~CEspLogAgentVariantIterator()
+{
+}
+
+bool CEspLogAgentVariantIterator::first()
+{
+    m_variantIt = variants().begin();
+    return isValid();
+}
+
+bool CEspLogAgentVariantIterator::next()
+{
+    if (m_variantIt != variants().end())
+        ++m_variantIt;
+    return isValid();
+}
+
+bool CEspLogAgentVariantIterator::isValid()
+{
+    return (m_variantIt != variants().end());
+}
+
+const IEspLogAgentVariant& CEspLogAgentVariantIterator::query()
+{
+    if (!isValid())
+        throw MakeStringException(0, "CEspLogAgentVariantIterator::query called in invalid state");
```

Comment:
I don't understand this comment. MakeStringException = makeStringExceptionV, MakeStringExceptionDirect = makeStringException, and there are thousands of occurrences MakeStringException. Would a better phrasing be that MakeStringExceptionDirect is preferred when no variable arguments are being used?

Reply:
There exist many MakeStringException s in our code. But, in new code, I have been asked to: a. change the MakeStringException(id, "") to makeStringException(id, ""); b.  change the MakeStringException(id, "...", ...) to makeStringExceptionV(id, "", ...). Please see https://github.com/hpcc-systems/HPCC-Platform/pull/14491#discussion_r565166491.

Reply:
It's an ongoing attempt to make our code more consistent, and use consistent naming schemes.. i.e. the standard "V" when a format string is provided, rather than the non-standard "Direct" when a format string is not provided.